### PR TITLE
Docs: Add which option is used by which ArangoBackup operation

### PR DESCRIPTION
**Pull request by Simran-B**
Add for which backup operation an option is meant

### CODE REVIEW

File: arangosh/Backup/BackupFeature.cpp [hunk 1]
```diff
@@ -681,68 +681,73 @@ void BackupFeature::collectOptions(std::shared_ptr<options::ProgramOptions> opti
...
 
-  options->addOption("--identifier", "a unique identifier for a backup",
+  options->addOption("--identifier",
+                     "a unique identifier for a backup "
+                     "(restore/upload/download operation)",
                      new StringParameter(&_options.identifier));
 
-  //  options->addOption("--include-search", "whether to include ArangoSearch data",
+  //  options->addOption("--include-search",
+  //                     "whether to include ArangoSearch data (??? operation)",
```
Comment (neunhoef) [inline arangosh/Backup/BackupFeature.cpp]: Does not matter, is not implemented, and we will have to revisit once we can backup ArangoSearch data.

File: arangosh/Backup/BackupFeature.cpp [hunk 2]
```diff
@@ -681,68 +681,73 @@ void BackupFeature::collectOptions(std::shared_ptr<options::ProgramOptions> opti
...
                      new DoubleParameter(&_options.maxWaitForLock));
 
   options->addOption(
       "--max-wait-for-restart",
-      "maximum time to wait (in seconds) for the server to restart after a "
+      "maximum time to wait in seconds for the server to restart after a "
       "restore operation before reporting an error; if zero, arangobackup will "
       "not wait to check that the server restarts and will simply return the "
-      "result of the restore request",
+      "result of the restore request (create operation)",
```
Comment (neunhoef) [inline arangosh/Backup/BackupFeature.cpp]: this must be "restore operation".

File: arangosh/Backup/BackupFeature.cpp [hunk 3]
```diff
@@ -681,68 +681,73 @@ void BackupFeature::collectOptions(std::shared_ptr<options::ProgramOptions> opti
...
       new DoubleParameter(&_options.maxWaitForRestart));
 
   options->addOption("--save-current",
                      "whether to save the current state as a backup before "
-                     "restoring to another state",
+                     "restoring to another state (create operation)",
```
Comment (neunhoef) [inline arangosh/Backup/BackupFeature.cpp]: This must be `restore operation`

File: arangosh/Backup/BackupFeature.cpp [hunk 4]
```diff
@@ -681,68 +681,73 @@ void BackupFeature::collectOptions(std::shared_ptr<options::ProgramOptions> opti
...
                      new BooleanParameter(&_options.abort));
 #endif
   /*
     options->addSection(
         "remote", "Options detailing a remote connection to use for operations");
 
     options->addOption("--remote.credentials",
-                       "the credentials used for the remote endpoint (see manual "
-                       "for more details)",
+                       "the credentials used for the remote endpoint. "
```
Comment (neunhoef) [inline arangosh/Backup/BackupFeature.cpp]: These are commented out and do not exist.

File: arangosh/Backup/BackupFeature.cpp [hunk 5]
```diff
@@ -681,68 +681,71 @@ void BackupFeature::collectOptions(std::shared_ptr<options::ProgramOptions> opti
...
 
   options->addOption("--save-current",
                      "whether to save the current state as a backup before "
-                     "restoring to another state",
+                     "restoring to another state (restore operation)",
                      new BooleanParameter(&_options.saveCurrent));
 #ifdef USE_ENTERPRISE
   options->addOption("--status-id",
-                     "returns the status of a transfere process",
+                     "returns the status of a transfere process "
```
Comment (kvahed) [inline arangosh/Backup/BackupFeature.cpp]: no 'e' at end of transfer.

File: arangosh/Backup/BackupFeature.cpp [hunk 6]
```diff
@@ -681,68 +681,71 @@ void BackupFeature::collectOptions(std::shared_ptr<options::ProgramOptions> opti
                      static_cast<std::underlying_type<Flags>::type>(Flags::Hidden));
 
   options->addOption("--allow-inconsistent",
-                     "whether to attempt to continue in face of errors (may "
-                     "result in inconsistent backup state)",
+                     "whether to attempt to continue in face of errors. "
```
Comment (kvahed) [inline arangosh/Backup/BackupFeature.cpp]: to be consistent, i'd suggest using semicolon here to or a dot below. my choice would be ... errors; may ...

**Comment by Simran-B** [comment]
transfere with an extra e occurs in actual code as well, @kvahed please check

**Pull request merged**